test/tests/materials/recompute/tests

```diff
@@ -0,0 +1,22 @@
+[Tests]
+  [./newton_iteration]
+    # Test the ability to compute material properties in an iterative manner
+    type = 'Exodiff'
+    input = 'recompute.i'
+    exodiff = 'recompute_out.e'
+  [../]
+
+  [./newton_no_calc]
+    # Test that no calculations are performed if computeProperties is not called explicitly on DiscreteMaterial
+    type = 'Exodiff'
+    input = 'recompute_no_calc.i'
+    exodiff = 'recompute_no_calc_out.e'
+  [../]
+
+  [./newton_error]
+    # Test error message when trying to get a non DiscreteMaterial object
+    type = 'RunException'
+    input = 'recompute_error.i'
+    expect_err = 'nable to locate the a DiscreteMaterial with the name recompute_props'
```

Comment:
Just curious about this expect_err. Are you skipping the first character because you don't want to worry about case? These are Regex's so you can just do `[Uu]nable to locate...`